### PR TITLE
linux-eic-shell.yml: enable on maintenance branches

### PR DESCRIPTION
**Pull request by veprbl**
Addresses an issue brought up in https://github.com/eic/EICrecon/pull/964#issuecomment-1720001570

### CODE REVIEW

File: .github/workflows/linux-eic-shell.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - 'v1.*' # maintenance branches
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:7]:
```suggestion
      - 'v*.*' # maintenance branches
```
?

Comment (veprbl) [reply .github/workflows/linux-eic-shell.yml:7]: Apparently, it even supports some regex-like https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
```suggestion
      - 'v[0-9]+.[0-9]+' # maintenance branches
```

Comment (wdconinc) [reply .github/workflows/linux-eic-shell.yml:7]: Yeah, I wasn't sure. I thought it just did limited globbing.

**Review by wdconinc — APPROVED** [review]

**Comment by epic-capybara** [comment]
Git push to origin failed for v1.5 with exitcode 1

**Pull request merged**